packages/util/src/environment.ts

```diff
@@ -72,6 +72,18 @@ export function isBrowser(): boolean {
   return typeof self === 'object' && self.self === self;
 }
 
+/**
+ * Detect browser extensions (Chrome and Firefox at least).
+ */
+declare const chrome: { runtime?: unknown };
+declare const browser: { runtime?: unknown };
+export function isBrowserExtension(): boolean {
+  return (
+    (typeof chrome === 'object' && chrome.runtime !== undefined) ||
```

Comment:
Just tested this condition in Chrome console on a random webpage (not in an extension) and it evaluates true, `chrome.runtime` exists.  I read somewhere that if there are any extensions on the page, even if you are not in the extension's code, this will happen? I guess it needs to be available on the page to pass messages from the content script to the background script.  [This suggests](https://stackoverflow.com/questions/7507277/detecting-if-code-is-being-run-as-a-chrome-extension) checking for `chrome.runtime.id` which is indeed undefined in my browser.

Reply:
Oh oops, saw you changed it already!

Reply:
Yup.  Just tested the new version in and out of an extension on Chrome and Firefox, and it seems to work.  Sorry for the earlier mistake!